File: Python/7py 2/max_rot.py
"""
|--------------------------------------------------------------------------
| Rotate for a Max
|--------------------------------------------------------------------------
|
"""

def max_rot(n):
   s, res = str(n), [n]
   for i in range(len(s)):
      s = s[:i] + s[i+1:] + s[i]
      res.append(int(s))
   return max(res)


def max_rot(n):
   res = [str(n)]
   for i in range(len(res[0])):
      res.append(res[-1][:i] + res[-1][i+1:] + res[-1][i])
   return int(max(res))


def max_rot(n):
   res,i,c= [],0,str(n)
   while i<len(str(n)):
      res.append(int(c))
      c= c[:i]+c[i+1:]+c[i]
      i+=1
   return max(res)


print(max_rot(38458215), 85821534)
print(max_rot(195881031), 988103115)
print(max_rot(896219342), 962193428)
print(max_rot(69418307), 94183076)

File: PHP/7php 2/maxRot.php
<?php
/*
|--------------------------------------------------------------------------
| Rotate for a Max
|--------------------------------------------------------------------------
|
*/

function maxRot($n) {
   $res[] = $n;
   $res[] = intval(substr($res[0], 1).substr($res[0], 0, 1));
   for ($i = 1; $i < strlen($n) - 1; $i++)
      $res[] = intval(substr($res[$i], 0, $i) . substr($res[$i], $i + 1) . substr($res[$i], $i, 1));
   return max($res);
}


function maxRot(int $n) {
   $str = $n . '';
   $sequence = [$n];
   foreach (range(0, strlen($str) - 2) as $i) {
      $str = substr($str, 0, $i) . substr($str, $i + 1) . $str[$i];
      $sequence[] = +$str;
   }
   return max($sequence);
}


function maxRot($n) {
   return max(array_reduce(range(0, strlen($n)), function($r, $i) {
      $last = $r[0];
      $strNumber = substr($last, 0, $i) . substr($last, $i+1) . substr($last, $i, 1);
      array_unshift($r, intval($strNumber));
      return $r;
   }, [$n]));
}


print_r(maxRot(56789) . ' ' . 68957 . '
');
print_r(maxRot(896219342) . ' ' . 962193428 . '
');